### ISSUE TITLE
Wrong use of `cargo_metadata` outside `build.rs`

### ISSUE THREAD

**Issue by virtualritz** [infobip-community/infobip-api-rust-sdk#15]
TLDR; if you build against this crate and copy the resulting executable anywhere else and run it from there, it will not work. Or uninstall your Rust toolchain and try to run it. It will not work.

Kindly pardon the snarky tone of what follows but this just cost me two hours.

I can't help but think that I must be the first person to actually use the Infobip Rust SDK in production-deployed code.

The fact that this crate sports two error variants called `SdkError::RustVersion` and `SdkError::CargoMetadata` seems to suggest that whoever wrote this was likely new to Rust.

## Details

Inside `src/api/mod.rs`, there is this gem:

```rust
fn get_user_agent() -> Result<String, SdkError> {
    let rust_version = version()?;
    let cmd = MetadataCommand::new();
    let package_version = cmd
        .exec()?
        .packages
        .into_iter()
        .find(|p| p.name == "infobip_sdk")
        .unwrap()
        .version;

    Ok(format!(
        "@infobip/rust-sdk/{} rust/{}",
        package_version, rust_version
    ))
}
```

* `rustc_version::version()` will horribly fail if you do not have a Rust compiler installed at *run time* (an assumption that is at best weird).

* `MetadataCommand::exec()` will fail horribly when used from outside *any* Rust project folder structure.

The `cargo_metadata` crate is meant for two use cases:

* *either* inside a `cargo` subcommand (which will typically only be ran from within a Rust project folder)

* *or* inside `build.rs`. I.e. to gather *static* data at *compile time*.

The same goes for the `rustc_version` crate.

*Neither* is meant to be used at *run time* outside of such contexts. Quote from the *top* of the `README` of `cargo_metadata` (emphasis mine):

> Structured access to the output of `cargo metadata`. **Usually used from within a `cargo-*` executable.**

And from the *top* of the README of `rustc_version` (emphasis mine):

> This can be used by **build scripts** or other **tools dealing with Rust *sources*** to make decisions based on the version of the compiler. 

## Reproduce

Either:
* Copy your SMS example program from `/target/debug/<name of the example>` to e.g. `$HOME` and run it from there.
* Just uninstall Rust/resp. make sure rustc is not in `$PATH` before trying to run.

For the former case you will get something like (this is the `?` in `sms_client.send(request_body).await?` unwrapping to an `Err`):

```
thread 'main' panicked at src/main.rs:29:56:
called `Result::unwrap()` on an `Err` value: CargoMetadata(CargoMetadata { stderr: "error: could not find `Cargo.toml` in `/home/moritz` or any parent directory\n" })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

## Fix

Now also included in #14.

**Comment by strosek** [comment]
Hi Moritz. I'm sorry these issues caused you to loose time. This crate is indeed one of our first Rust projects, but we are improving. Thanks for your contributions!

**Comment by virtualritz** [comment]
No worries, sorry I was bit grumpy. It was one of those days. :grin:

**Comment by virtualritz** [comment]
I'm re-opening as I just realized there is update published on `crates.io` yet.

All the latest published versions of this crate (`0.5.0`, `0.5.1` and `0.5.2`) on are completely broken for actual production use (i.e. beyond running some examples locally) because of this issue.

I.e. you should at least yank *any previous `5.x` version* from crates.io and ideally also publish `0.5.3` with a fix for this, cherry-picked from #14).
